crates/sui-framework/packages/dwallet-system/sources/dwallet_2pc_mpc_ecdsa_k1.move

```diff
@@ -0,0 +1,302 @@
+        messages: vector<vector<u8>>,
+        sender: address,
+    }
+
+    struct NewSignDataEvent has store, copy, drop {
+        presign_session_id: ID,
+        hash: u8,
+        dkg_output: vector<u8>,
+        public_nonce_encrypted_partial_signature_and_proofs: vector<u8>,
+        presigns: vector<u8>,
+    }
+
+    // <<<<<<<<<<<<<<<<<<<<<<<< Events <<<<<<<<<<<<<<<<<<<<<<<<
+
+
+    #[allow(unused_field)]
+    struct DWallet has key, store {
+        id: UID,
+        session_id: ID,
+        dwallet_cap_id: ID,
+        output: vector<u8>,
+        public_key: vector<u8>,
+    }
+
+    struct DKGSession has key {
+        id: UID,
+        dwallet_cap_id: ID,
+        commitment_to_centralized_party_secret_key_share: vector<u8>,
+        sender: address,
+    }
+
+    struct DKGSessionOutput has key {
+        id: UID,
+        session_id: ID,
+        dwallet_cap_id: ID,
+        commitment_to_centralized_party_secret_key_share: vector<u8>,
+        secret_key_share_encryption_and_proof: vector<u8>,
+    }
+
+    #[allow(unused_field)]
+    struct PresignSession has key {
+        id: UID,
+        dwallet_id: ID,
+        dwallet_cap_id: ID,
+        hash: u8,
+        commitments_and_proof_to_centralized_party_nonce_shares: vector<u8>,
+        messages: vector<vector<u8>>,
+        sender: address,
+    }
+
+    #[allow(unused_field)]
+    struct PresignSessionOutput has key {
+        id: UID,
+        session_id: ID,
+        dwallet_id: ID,
+        dwallet_cap_id: ID,
+        output: vector<u8>,
+    }
+
+    #[allow(unused_field)]
+    struct Presign has key {
+        id: UID,
+        session_id: ID,
+        dwallet_id: ID,
+        dwallet_cap_id: ID,
+        presigns: vector<u8>,
+    }
+
+    #[allow(unused_field)]
+    struct SignData has store {
+        presign_session_id: ID,
+        hash: u8,
+        public_nonce_encrypted_partial_signature_and_proofs: vector<u8>,
+        presigns: vector<u8>,
+    }
+
+    public fun create_dkg_session(commitment_to_centralized_party_secret_key_share: vector<u8>, ctx: &mut TxContext): DWalletCap {
+        let cap = create_dwallet_cap(ctx);
+        let sender = tx_context::sender(ctx);
+        let session = DKGSession {
+            id: object::new(ctx),
+            dwallet_cap_id: object::id(&cap),
+            commitment_to_centralized_party_secret_key_share,
+            sender,
+        };
+        event::emit(NewDKGSessionEvent {
+            session_id: object::id(&session),
+            dwallet_cap_id: object::id(&cap),
+            commitment_to_centralized_party_secret_key_share,
+            sender,
+        });
+        transfer::freeze_object(session);
+        cap
+    }
+
+    #[allow(unused_function)]
+    fun create_dkg_output(session: &DKGSession, commitment_to_centralized_party_secret_key_share: vector<u8>, secret_key_share_encryption_and_proof: vector<u8>, ctx: &mut TxContext) {
+        assert!(tx_context::sender(ctx) == @0x0, ENotSystemAddress);
```

Comment:
I'm not sure what this assert is for, but shouldn't it be for `create_dkg_session()` and many others in case it is actually needed here?

Reply:
This function should be called only by the system, and this is what we check for